src/NuGet.Clients/NuGet.Console/Console/ConsoleDispatcher.cs

```diff
@@ -158,10 +159,12 @@ private void RaiseEventSafe(EventHandler handler)
         public void Start()
         {
             // Only Start once
-            lock (_lockObj)
-            {
-                if (_dispatcher == null)
+            LazyInitializer.EnsureInitialized(ref _dispatcher,
```

Comment:
Same as the other one.

Reply:
@nkolev92 there is no `AsyncLazy` version of this `LazyInitializer.EnsureInitialized`. Do we just abandon this PR?

Reply:
I need to do some tests but I haven't had the time to dig in. I'll try to do that soon so I unblock this PR. 
Sorry for the delays, there are a bunch of higher pri PRs that I had to look at (ones affecting 5.7)

Reply:
@nkolev92 We need to make decision on this PR. It has been here for 2 months. Maybe just abandon it since there is no actual need or urgency of implement this change. I feel it's nice to have thing.

Reply:
> there is no AsyncLazy version of this LazyInitializer.EnsureInitialized

Consider the semantic meaning of `LazyInitializer.EnsureInitialized` and how that behavior differs from `AsyncLazy`. Aren't they the same?

Reply:
> also I don't see we really need this change. Do we just abandon this PR?

I support this change because it avoids locking the object every time `Start()` method is invoked even though `value` has been initialized already. As per my understanding, the [`overloaded method of EnsureInitialized`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.lazyinitializer.ensureinitialized?view=netcore-3.1#System_Threading_LazyInitializer_EnsureInitialized__1___0__System_Boolean__System_Object__System_Func___0__) invoked here is thread safe. [LazyIntializer doc](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization).

Reply:
Left a comment on the other thread as well, when I posted the original comment I was looking at the wrong description.

My point about a potential deadlock still stands. Lazy initializer won't play well when the UI thread is blocked on the initialization, yet internally the initalization step needs the UI thread.